flocker/node/agents/blockdevice.py

```diff
@@ -1511,6 +1513,16 @@ def _mountpath_for_dataset_id(self, dataset_id):
     def calculate_changes(self, configuration, cluster_state):
         this_node_config = configuration.get_node(
             self.node_uuid, hostname=self.hostname)
+        local_state = cluster_state.get_node(self.node_uuid,
+                                             hostname=self.hostname)
+
+        # We need to know applications (for now) to see if we should delay
+        # deletion or handoffs. Eventually this will rely on leases instead.
+        if local_state.applications is None:
+            return in_parallel(changes=[])
```

Comment:
Was there another issue filed for this part?  It's pretty related, so maybe not... But it'd be nice to make sure it gets closed at the same time if so.